sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InMemoryFileIndex.scala

```diff
@@ -147,32 +142,6 @@ class InMemoryFileIndex(
 
 object InMemoryFileIndex extends Logging {
 
-  /** A serializable variant of HDFS's BlockLocation. */
-  private case class SerializableBlockLocation(
-      names: Array[String],
-      hosts: Array[String],
-      offset: Long,
-      length: Long)
-
-  /** A serializable variant of HDFS's FileStatus. */
-  private case class SerializableFileStatus(
```

Comment:
Looks like it's not just simply moving the codes but includes code refactoring. Can we separate PRs? one for refactoring and the other one for moving the codes.

This listing is very critical to Spark SQL and other APIs. I would like to be very cautious on touching it.

Reply:
Yeah good suggestion. I'll limit this PR to only refactoring then. Should I create another JIRA for the cleanup PR? I'm pretty new to Spark :)

Reply:
Yeah a seperate JIRA is best.